architecture/event_processor.md

```diff
@@ -0,0 +1,137 @@
+# Analytic Event Processor
+
+The Event Processor is responsible for consuming, batching, and delivering events generated
+by the server and client-side LaunchDarkly SDKs.
+
+```mermaid
+classDiagram
+    IEventProcessor <|-- NullEventProcessor
+    IEventProcessor <|-- AsioEventProcessor
+
+    AsioEventProcessor *-- LRUCache
+    AsioEventProcessor *-- Outbox
+    AsioEventProcessor *-- WorkerPool
+    AsioEventProcessor *-- Summarizer
+
+    RequestWorker *-- "1" EventBatch
+    WorkerPool *-- "5" RequestWorker
+
+    TrackEvent -- TrackEventParams
+    InputEvent *-- FeatureEventParams
+    InputEvent *-- IdentifyEventParams
+    InputEvent *-- TrackEventParams
+
+    OutputEvent *-- IndexEvent
+    OutputEvent *-- FeatureEvent
+    OutputEvent *-- DebugEvent
+    OutputEvent *-- IdentifyEvent
+    OutputEvent *-- TrackEvent
+
+    IEventProcessor <-- InputEvent
+    Outbox <-- OutputEvent
+
+    class IEventProcessor {
+        <<interface>>
+        +SendAsync(InputEvent event) void
+        +FlushAsync() void
+        +ShutdownAsync() void
+    }
+
+    class NullEventProcessor {
+
+    }
+
+    class AsioEventProcessor {
+
+    }
+
+    class EventBatch {
+        +const Count() size_t
+        +const Request() network:: HttpRequest
+        +const Target() std:: string
+    }
+
+    class LRUCache {
+        +Notice(std:: string value) bool
+        +const Size() size_t
+        +Clear() void
+    }
+
+    class Outbox {
+        +PushDiscardingOverflow(std:: vector~OutputEvent~ events) bool
+        +Consume() std:: vector events
+        +const Empty() bool
+    }
+
+    class RequestWorker {
+        +const Available() bool
+        +AsyncDeliver(EventBatch, delivery_callback)
+    }
+
+    class WorkerPool {
+        +Get(worker_callback) void
+    }
+
+    class Summarizer {
```

Comment:
It seems like, in the implementation, the summarizer is also the data representation for the summary event? Maybe needs some means to indicate that, because I initially thought we were just missing summary events after looking at the diagram.

Reply:
That's a good point. Perhaps I should just add a `SummaryEvent` to make this intuitive, it wouldn't really change anything - let me try it.

Reply:
The change didn't really make sense. I'll document this instead in the diagram.